ompi/mca/mtl/ofi/mtl_ofi.h

```diff
@@ -676,14 +703,27 @@ ompi_mtl_ofi_send_excid(struct mca_mtl_base_module_t *mtl,
                   bool is_send)
 {
     ssize_t ret = OMPI_SUCCESS;
-    ompi_mtl_ofi_request_t *ofi_req = malloc(sizeof(ompi_mtl_ofi_request_t));
+    ompi_mtl_ofi_request_t *ofi_req = NULL;
     int ctxt_id = 0;
-    mca_mtl_ofi_cid_hdr_t *start = malloc(sizeof(mca_mtl_ofi_cid_hdr_t));
+    mca_mtl_ofi_cid_hdr_t *start = NULL;
     ompi_proc_t *ompi_proc = NULL;
     mca_mtl_ofi_endpoint_t *endpoint = NULL;
     fi_addr_t sep_peer_fiaddr = 0;
     mca_mtl_comm_t *mtl_comm;
     
+    ofi_req = (ompi_mtl_ofi_request_t *)malloc(sizeof(ompi_mtl_ofi_request_t));
+    if (NULL == ofi_req) {
+        ret =  OMPI_ERR_OUT_OF_RESOURCE;
+        goto fn_exit;
+    }
+
+    start = (mca_mtl_ofi_cid_hdr_t *)malloc(sizeof(mca_mtl_ofi_cid_hdr_t));
+    if (NULL == start) {
+        free(ofi_req);
```

Comment:
This isn't needed since fn_exit does the cleanup anyway

Reply:
good point i'll remove that free